src/Microsoft.AspNetCore.Blazor.Razor.Extensions/ComplexAttributeContentPass.cs

```diff
@@ -73,17 +74,59 @@ private static bool TrySimplifyContent(IntermediateNode node)
                 //
                 // There's a special case here for explicit expressions. See https://github.com/aspnet/Razor/issues/2203
                 // handling this case as a tactical matter since it's important for lambdas.
-                if (cSharpNode.Children.Count == 3 &&
-                    cSharpNode.Children[0] is IntermediateToken token0 &&
-                    cSharpNode.Children[2] is IntermediateToken token2 &&
-                    token0.Content == "(" &&
-                    token2.Content == ")")
                 {
-                    cSharpNode.Children.RemoveAt(2);
-                    cSharpNode.Children.RemoveAt(0);
+                    if (cSharpNode.Children.Count == 3 &&
```

Comment:
Reminder to myself to recompute the span of these nodes after editing them. I think some of the issues related to completion are due to these spans being wrong.